src/Adapter/Hosting/HostingInformation.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * 2007-2017 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2017 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+namespace PrestaShop\PrestaShop\Adapter\Hosting;
+
+use Tools;
+use Db;
+
+/**
+ * Class HostingInformation
+ *
+ * Depends on Db legacy class right now, needs to be migrated using Doctrine.
+ * This class should not be used outside of PrestaShop context.
+ */
+class HostingInformation
+{
+    /**
+     * @return array
+     */
+    public function getDatabaseInformation()
+    {
+        return array(
+            'version' => Db::getInstance()->getVersion(),
+            'server' => _DB_SERVER_,
+            'name' => _DB_NAME_,
+            'user' => _DB_USER_,
+            'prefix' => _DB_PREFIX_,
+            'engine' => _MYSQL_ENGINE_,
+            'driver' => Db::getClass(),
+        );
+    }
+
+    /**
+     * @return array
+     */
+    public function getServerInformation()
+    {
+        return array(
+            'version' => $_SERVER['SERVER_SOFTWARE'],
+            'php' => $this->getPhpInformation(),
+        );
+    }
+
+    /**
+     * @return array
+     */
+    public function getPhpInformation()
+    {
+        return array(
+            'version' => phpversion(),
+            'memoryLimit' => ini_get('memory_limit'),
+            'maxExecutionTime' => ini_get('max_execution_time'),
+            'maxFileSizeUpload' => ini_get('upload_max_filesize')
+        );
+    }
+
+    /**
+     * @return string
+     */
+    public function getUname()
+    {
+        return function_exists('php_uname') ? php_uname('s').' '.php_uname('v').' '.php_uname('m') : '';
```

Comment:
Was wondering the same, and why this check was done on PrestaShop on the first place: it appears that PHP is not always compiled with PHP, so `php_uname` can't exists.

Reply:
> PHP is not always compiled with PHP

🙃

Reply:
:D `php_uname` function is not always compiled with PHP

Reply:
I couldn't find that anywhere, where did you see it? 🤔

Reply:
I don't remember, searching on google I think the main issue is that [some hosters](https://www.google.fr/search?q=php_uname+has+been+disabled&oq=php_uname+has+been+disabled&gs_l=psy-ab.3..0i22i30k1l4.5845.6063.0.6223.2.2.0.0.0.0.135.207.1j1.2.0.foo%2Cnso-ehuqi%3D1%2Cnso-ehuui%3D1%2Cewh%3D0%2Cnso-mplt%3D2%2Cnso-enksa%3D0%2Cnso-enfk%3D1%2Cnso-usnt%3D1%2Cnso-qnt-npqp%3D0-1701%2Cnso-qnt-npdq%3D0-54%2Cnso-qnt-npt%3D0-1%2Cnso-qnt-ndc%3D300%2Ccspa-dspm-nm-mnp%3D0-05%2Ccspa-dspm-nm-mxp%3D0-125%2Cnso-unt-npqp%3D0-17%2Cnso-unt-npdq%3D0-54%2Cnso-unt-npt%3D0-0602%2Cnso-unt-ndc%3D300%2Ccspa-uipm-nm-mnp%3D0-007525%2Ccspa-uipm-nm-mxp%3D0-052675...0...1.1.64.psy-ab..0.2.204.e89kXZ93ULY) disable `php_uname` for security reasons: don't know why this can be a security issue.

Anyways, I don't want to risk a BC on it.

Reply:
Huh, I only just discovered that there's a [php.ini directive](http://php.net/manual/en/ini.core.php#ini.disable-functions) that can be used to disable any PHP function (and there's one for classes, too).

So I guess we can leave it as is.